src/ddsts/include/dds/ddsts/typetree.h

```diff
@@ -72,7 +76,7 @@ typedef char *ddsts_identifier_t;
  * IDL elements, such as the constant definition and the case labels.
  */
 
-typedef uint32_t ddsts_flags_t;
+typedef uint64_t ddsts_flags_t;
```

Comment:
Yes, I agree with you, that with respect to the types, it is not really used as a bitmask. (Note that it is a bitmask in combination with DDSTS_REFERENCE_1 and DDSTS_REFERENCE_2.) I see that some bits could be saved by, for example, by adding a flag for FORWARD and combine FORWARD_STRUCT and FORWARD_UNION as a combination of FORWARD and respectively STRUCT and UNION. Note that this would save just a single bit (add one to remove two). With the number of grammar elements still to be added, I am afraid that 32 bits is not going to be enough anyway. At the moment, the code implements a union of types with a bitwise union, which would be no longer possible if a bitmask would be used (of course, depending on how the bitmask is organized). Anyway, if we would create a bitmask for the types and it would still fit inside 32 bits, when all grammar element are added (or change it into an enum), we would still need another bitmask for the references. A uint8_t could be used for that. But on each compiler that implements a 32-alignment for struct members that would not reduce the memory footprint for tree nodes.